### DISCUSSION TITLE
Alarm control panel feedback

### DISCUSSION THREAD

**Discussion by edwardtfn** [Blackymas/NSPanel_HA_Blueprint#984] [General]
Support to Alarm Control Panel is available on `dev` and we would appreciate if you can give some early feedback, so we can adjust it before going to a wider public.
Feel free to give any kind of feedback on the user interface, behavior and documentation.

Docs: https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/dev/docs/en/alarm.md
~~Release notes: https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/dev/ReleaseNotes.md#2-alarm-control-panel~~

For reference, some of this discussion started on #930.

**Comment by primeroz** [comment]
In my case, when using an `esp32` alarm panel using the [`alarm_control_panel esphome component`](https://esphome.io/components/alarm_control_panel/index.html)

Few things to think about : 
---
1. I have an `arming delay` between the moment the `arm_XXX` button is pushed  and the moment moment when the state changes to `armed_xxx` . during this `ARMING` time  all buttons are gray , ideally it would be `yellow` until it turns green 

see this video for example of behaviour.
https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/2085772/3a01ea97-70b5-4ebc-883d-710768eeabf8

Also there should be a `sound feedback` using the buzzer when the alarm is `ARMING` , similar to the old school alarm panel that will beep until the alarm is actually armed to give time to exit the house.

---
2. When the alarm is `Triggered` ( or `Pending` ) there should be ideally a `visual` and a `sound` feedback ( or both ) . This is particularly useful when entering the house to be alerted that the alarm is `PENDING` and will `TRIGGER` if not disarmed. 

---
3. It would be great if the `home page` icon could represent the state of the alarm so `what type of ARMED_XXXX` is enabled and a color of orange for triggered and red for triggered

**Comment by edwardtfn** [comment]
I wanna go a bit deeper on item 2:

> When the alarm is Triggered ( or Pending ) there should be ideally a visual and a sound feedback ( or both ) . This is particularly useful when entering the house to be alerted that the alarm is PENDING and will TRIGGER if not disarmed.

Let's say the panel is sleeping when the entity changes from `armed_away` to `triggered`...
Should the panel wake up?
And if so, what should be shown? Home page or Alarm page?
If Alarm page is shown (automatically or manually), should the `Away` button be `red`? What about the other buttons?

**Reply by primeroz** [reply]
For my personal use case or my ideal use case I was thinking something along those lines. Of course this is just a suggestion ( and i am partly trying to implement it right now just using home assistant automations) . Also i understand some of the following might be tricky to implement so is really just for discussion 

```mermaid
flowchart TD
    A[DISARMED] 
    B[ARMING]
    C[ARMED_NIGHT]
    D[PENDING]
    E[TRIGGERED]
    F[ARMED_AWAY]
    G[ARMED_XXX]
    A -->|Buzzer beeping\nButton Amber\nWake up from sleep or Dim| B
    B --> C
    B --> F 
    B --> G 
    C -->|Alarm Fire\nWake up from sleep or Dim| D 
    F -->|Alarm Fire\nWake up from sleep or Dim| D 
    G -->|Alarm Fire\nWake up from sleep or Dim| D 
    D -->|Buzzer beeping\nButton Red\nWake up from sleep or Dim| E
  ```

> Let's say the panel is sleeping when the entity changes from armed_away to triggered...
> Should the panel wake up?

IMHO yes, note that it will go through the pending state first but for me yes.

In a standard home alarm system the panel is more than just a keypad to turn the alarm on and off. It is a quick visual and sound feedback to what the situation is like. 

So in my opinion yes , it should wake up from sleep or increase brightness if dimmed and possibly even use the buzzer to beep for a while ( as long as the `ARMING` or `PENDING` state is live even though that might be tricky )  

> And if so, what should be shown? Home page or Alarm page?

Ideally `Alarm page` , alternatively `Home page` with a visual feedback like `red icon` for the alarm ... but i think the `Alarm page` would be better. similar to how the `notification` will wake up the panel on the `notification page` and not the `home page` 

> If Alarm page is shown (automatically or manually), should the Away button be red? What about the other buttons?

This of course will depend a lot on ease of implementation.

If possible i think it would be ideal to have the `armed_XXX` that was set turn RED and the rest gray, but from my understanding when the `alarm_control_panel` resource will change `state` to `PENDING` or `TRIGGERED` there will be no context anymore about which `armed__XXX` state it was on ... so in order to light up only one button you might need to store it somewere in a var in the nextion code ? 

Also a dedicated page with `TRIGGERED` label / graphic and a `DISARM` button could work ? that is how i was implementing a very simple dedicated nspanel alarm very similar to https://github.com/formatBCE/NSPanel-simple-alarm 

Something like 

```mermaid
flowchart TD
    A[Home Page] 
    B[Alarm Page\nDisarmed Button Green]
    C[Alarm Page\nArmed_XXX Button Green]
    D[Alarm Pending Page\nDisarm Button Only]
    E[Alarm Triggered Page\nDisarm Button Only]

    A -->|"`Click on Icon`"| id1(["Switch to Alarm page"]) --> B 
    B -->|"Click on ARM button"| id3(["Keypad Popup"]) -->|"ARMED_XXX Button turns Amber\nBuzzer Beeping"| id2(["After Arming Delay\n"]) --> C 

    idEvent{{"External Alarm Event"}} -->|"Wake up from Sleep or Dim"| id4(["Switch to dedicated Pending Page"]) -->|Buzzer Beeping| D 
    D -->|After Pending Delay| E 
  
```

**Reply by edwardtfn** [reply]
> If possible i think it would be ideal to have the armed_XXX that was set turn RED and the rest gray, but from my understanding when the alarm_control_panel resource will change state to PENDING or TRIGGERED there will be no context anymore about which armed__XXX state it was on ... so in order to light up only one button you might need to store it somewere in a var in the nextion code ?

I have the current status stored in a temporary variable at ESPHome, so when the service to change that is called, I have both values for a moment and that could be used.
Also, on the Blueprint side, when the state changes and triggers the update of alarm data, I have both `to_state` and `from_state`, so again I could use this to define which button would be `red`.

**Reply by edwardtfn** [reply]
You came with another important point... Should "Disarm" be the only option when the alarm is triggered? Or should be possible to click in one of the arm buttons to stop the trigger but the rearm the alarm?
Do you know if HA supports going directly from `triggered` to `armed_xxxx`?

I don't have a real alarm at home so I'm limited to play ith virtual alarms, which is not the same.

**Reply by edwardtfn** [reply]
Also, when you say "buzzer beeping", what kind of sound ate you expecting? Just a click or something continously buzzing?

**Reply by primeroz** [reply]
> I have the current status stored in a temporary variable at ESPHome, so when the service to change that is called, I have both values for a moment and that could be used.
Also, on the Blueprint side, when the state changes and triggers the update of alarm data, I have both to_state and from_state, so again I could use this to define which button would be red.

Perfect :+1:

> Should "Disarm" be the only option when the alarm is triggered? Or should be possible to click in one of the arm buttons to stop the trigger but the rearm the alarm?
> Do you know if HA supports going directly from triggered to armed_xxxx?


In many cases the only transition allowed from an `ARMED state` or a `TRIGGERED state` is to `disarmed` . 
You cannot , for example, go directly from `armed_night` to `armed_away` , you always have to pass through a `disarmed state` 

I think this is `implementation specific` but in the case of `alarm_control_panel esphome component` that is the case , see https://github.com/esphome/esphome/blob/dev/esphome/components/alarm_control_panel/alarm_control_panel_call.cpp#L66 

Basically from ARMED, PENDING or TRIGGERED you can only go to DISARMED

Also, something very common, is that `after a triggered timeout` the triggered state tend to go back into the last `armed state` and the alarm is `cleared` until a new event 

for example https://esphome.io/components/alarm_control_panel/template 
```
trigger_time (Optional, [Time](https://esphome.io/guides/configuration-types#config-time)): The time after a triggered alarm before resetting to previous state if the sensors are cleared/off. Defaults to 0s.
```

> Also, when you say "buzzer beeping", what kind of sound ate you expecting?

Sorry if i was not very clear, been writing this while cooking dinner. 
What i am thinking of is a situation like this 


* Alice is leaving her house
* NSPanel is mounted near the exit door
* She set `ARM_AWAY` 
* The exit delay before the alarm is armed to away mode starts 
  * the Buzzer start `beeping` to indicate that `alarm is ARMING` 
  * this is the time, normally 10-30 seconds , to allow people to exit without triggering the actual siren
---
* Bob enter the house , Armed in away mode
* A sensor trigger the alarm
* The alarm goes into Pending state
*  The entry delay before the alarm is triggered starts
    * The buzzer start beeping to indicate the alarm is `Pending` 
    * this is the time, normally 10-30 seconds, to allow people to disarm the alarm before the siren kicks in 

>  Just a click or something continously buzzing?

My current, old school, alarm beeps "On off On off" but it does not really matter i guess, something not too obnoxious and possibly with an `enable buzzer` flag in the blueprint ? 

> I don't have a real alarm at home so I'm limited to play ith virtual alarms, which is not the same.

If you have any `esp32` microcontroller hanging around i could help you build a `real` alarm very easily ... basically you just need something like https://gist.github.com/primeroz/86580a586b3c089327c5f620addb9b78 using some `fake` sensors from home assistant to simulate PIR and DOORs

**Comment by fzambroni** [comment]
So far, the Alarm Panel is working very well. 

https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/52335639/61d552e0-b735-4252-b132-0e63e3f18ba7

**Comment by Doughboy68** [comment]
Can there be a status on the home page indicating state of alarm eg. ready (green), not ready (yellow), armed (red), maybe changing the color of the icon in lower right? 

When I arm from home assistant it displays pending during the exit delay.

Also a way to view/bypass zones. This will be different depending on the alarm panel and not really possible/easy to figure out how many zones and display them all, but an optional entity or button page that we can hardcode in blueprint to be called from the alarm panel to view/bypass - similar to how we setup the entities page.


The envisalink integration has entities for each zone's status, and also additional bypass entities for each zone that can be toggled eg. leave a window open and want to still arm the house. I am sure other alarms have similar.

Here is a screenshot of my home assistant and how I display the zones and pressing them toggles. I think may be more difficult to do this in nspanel since it uses templates, so instead two separate screens: one to view zones, and another which toggles the bypass.

![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/16811687/a46d0314-4c32-4428-a978-0d459bd5e708)

This alarm type typically have 8 zones minimum and maybe 16 at most. Would be interesting to hear how many zones people have. If greater than a page of 8, would just populate the blueprint entities with most common ones would want to bypass (motion sensor, front door etc. would probably not be things you would normally bypass).

Can do this now with an entity page and a button page. and just be able to call them from the alarm screen. 
Here is entities of zones I care about...
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/16811687/9e373935-c0a9-46e8-aa9b-08df56f3a70e)

And here is a button page to toggle bypass...

![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/16811687/abf90452-671b-4018-b1ac-7c1dc3cd9ca7)

Note I didn't open the basement rec window for this example, but if was actually open it would show on instead of off (not sure why doesn't use correct open/close in nspanel), then would just know to go and bypass that zone and then be able to arm the alarm.

Edit... after typing all this, I looked closer and there is a grayed out Custom Byp button in lower left so maybe that is the plan for this? Or is this for supported alarms, if so and mine is not suported, it would be nice to have this button go to a page where can swipe between an entity screen and bypass button screen as above.

**Reply by edwardtfn** [reply]
> Can there be a status on the home page indicating state of alarm eg. ready (green), not ready (yellow), armed (red), maybe changing the color of the icon in lower right?

The possible states are:


Value | Description
-- | --
disarmed | The alarm is disarmed (off).
armed_home | The alarm is armed in home mode.
armed_away | The alarm is armed in away mode.
armed_night | The alarm is armed in night mode.
armed_vacation | The alarm is armed in vacation mode.
armed_custom_bypass | The alarm is armed in bypass mode.
pending | The alarm is pending (towards triggered).
arming | The alarm is arming.
disarming | The alarm is disarming.
triggered | The alarm is triggered.

Which should be red, amber or green?

**Reply by Doughboy68** [reply]
Here are icons and color suggestions:

disarmed - white/grey  mdi:shield-off-outline

armed_ all modes - green with appropriate icon:
  home - green mdi:shield-home-outline
  away - green mdi:shield-lock-outline
  night - green mdi:shield-moon-outline
  vacation - green mdi:shield-airplane-outline
  custom_bypass  - green mdi:shield-half-full

pending - probably not necessary as this should happen instant.

arming - amber mdi:shield-outline

disarming - probably not necessary as this should happen instant.

triggered - red mdi:shield-alert-outline (not 100%  because I didn't trigger mine but this combination makes sense)

**Reply by edwardtfn** [reply]
> Can do this now with an entity page and a button page. and just be able to call them from the alarm screen.

How is the flow when you are setting this from Home Assistant? And do your alarm system have it's own display (or an app)? How is the flow there for selecting custom bypass?

**Comment by edwardtfn** [comment]
What about when the alarm is unknown?

For now, I will set the home page icon like this:

State | Color | Icon | Icon name | Description
-- | -- | -- | -- | --
disarmed | White |![shield-off-outline](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/94725493/5b1c8219-3958-4096-9100-0b4ef47e7dac)|mdi:shield-off-outline | The alarm is disarmed (off).
armed_home | Green |![shield-home-outline](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/94725493/1ed9347b-139a-4729-a795-b08934b56581)| mdi:shield-home-outline | The alarm is armed in home mode.
armed_away | Green |![shield-lock-outline](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/94725493/998db33c-3ec5-47dd-9686-c28eee81e9eb)| mdi:shield-lock-outline | The alarm is armed in away mode.
armed_night | Green |![shield-moon-outline](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/94725493/84528bca-577d-4a43-8366-e6f9e41bc7b4)| mdi:shield-moon-outline | The alarm is armed in night mode.
armed_vacation | Green |![shield-airplane-outline](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/94725493/067218de-ec8c-4fce-a427-7848a7ae9f27)| mdi:shield-airplane-outline | The alarm is armed in vacation mode.
armed_custom_bypass | Green |![shield-half-full](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/94725493/af7ed346-28eb-4796-b62b-2a51e9cb6272)| mdi:shield-half-full | The alarm is armed in bypass mode.
pending | Amber |![shield-outline](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/94725493/31bcd613-04d6-455a-a443-3299211d487e)| mdi:shield-outline | The alarm is pending (towards triggered).
arming | Amber |![shield-outline](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/94725493/31bcd613-04d6-455a-a443-3299211d487e)| mdi:shield-outline | The alarm is arming.
disarming | Amber |![shield-off-outline](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/94725493/37c02584-dd65-4575-8d95-c7346de599e4)| mdi:shield-off-outline | The alarm is disarming.
triggered | Red |![shield-alert-outline](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/94725493/ee1ec55b-e671-4aa0-9baa-77eb2a8c8c7f)| mdi:shield-alert-outline | The alarm is triggered.
unknown / unavailable | White |![shield-alert-outline](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/94725493/a382e5ab-2e16-453c-bc9b-b3dcd3825de7)| mdi:shield-alert-outline | The alarm state is unknown or unavailable

**Reply by edwardtfn** [reply]
Done!
You have to update ESPHome and BluePrint.

**Reply by doron1** [reply]
Umm, I just updated to this version, and now the alarm icon on the home screen reflects the state of the alarm (disarmed) - that's good - but when I push it, all the elements/buttons are greyed out with text that's not readable. What caused this?

**Comment by fzambroni** [comment]
Same issue here, solved after restart my server.

**Reply by doron1** [reply]
Restarted HA, no change :-(

**Comment by fzambroni** [comment]
Make sure that you are using the last blueprint file.

**Reply by edwardtfn** [reply]
That's right. The call for the alarm function changed, so it will fail when calling the alarm page if the Blueprint isn't updated.

**Comment by doron1** [comment]
Of course, I was just being stupid. After updating the blueprint it now works well.

Follow up question: Is it possible for the blueprint to interrogate the alarm system, see whether a PIN is required, and ask for it only if it is indeed required?
Ideally, it will check for two attributes - "PIN required for arm" and "PIN required for disarm", and pop up the keyboard in each case as required.

**Reply by Doughboy68** [reply]
Great I will try when home.

Regarding my other suggestions regarding making a dedicated bypass page and entity page listing zone status?A custom page(s) could also be designed to combine both status and a bypass toggle - maybe can be in next version depending on release plans?

**Reply by edwardtfn** [reply]
> Follow up question: Is it possible for the blueprint to interrogate the alarm system, see whether a PIN is required, and ask for it only if it is indeed required?
Ideally, it will check for two attributes - "PIN required for arm" and "PIN required for disarm", and pop up the keyboard in each case as required.

I will look at that. It's certainly possible.

**Reply by edwardtfn** [reply]
> Regarding my other suggestions regarding making a dedicated bypass page and entity page listing zone status?A custom page(s) could also be designed to combine both status and a bypass toggle - maybe can be in next version depending on release plans?

Most likely will have to wait for following version, but I'd like to look at this sooner than later, but as I don't have an alarm to test, it would be nice if someone could share a short video showing how it works with Home Assistant or other system, so I can understand better the use flow.

**Reply by Doughboy68** [reply]
I discovered something else about HA alarms... mine anyways. The main status of the alarm (in my case "Home Alarm") seems to have no state whether it is ready to arm or not. In my alarm system, there is an additional sensor entity named keypad which is normally "ready" but will change to "not ready" when I open a door or window for example, and the physical keypad mounted on wall displays same via an LED. In this not ready state, the alarm is not able to be alarmed unless the opened zones are closed, or are set to bypass. Meanwhile, the alarm state still remains as disarmed.

Here is a screenshot to get an idea what I mean... there are bypass toggles for each zone, and then the sensor entities display the current status of each zone. In the screenshot of my previous post, I combine these using template so I can toggle on/off the bypass while displays the actual sensor. I don't expect this to happen on NSPanel, and two separate screens is fine for now. While the envisalink is fairly popular, I am interested in other alarms and how bypass and ready to arm status is handled. Don't want to make it work just for my system, while not working for others properly. Maybe there is a better way once finding out more information.

![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/16811687/2e11492a-ed1e-4ff5-81d4-1e65c4c851c3)

So the easy solution for me is to just add that keypad entity to proposed status/entity page that I would populate myself via blueprint.

Here is a video I found with my keypad which shows the arm/disarm and the ready status.
https://www.youtube.com/watch?v=5khUTTPcw2s

**Reply by edwardtfn** [reply]
>Follow up question: Is it possible for the blueprint to interrogate the alarm system, see whether a PIN is required, and ask for it only if it is indeed required?
Ideally, it will check for two attributes - "PIN required for arm" and "PIN required for disarm", and pop up the keyboard in each case as required.

I've added support to this, but again, I couldn't test.
Could you please let me know if this is working as expected for you?

You will have to update TFT, ESPHome and Blueprint.

**Reply by doron1** [reply]
You are faster than our combined ability to come up with suggestions! Amazing.
Just tested it. Works well. Indeed you implemented it only for arm and not for disarm - I can see why (HA only exposes "code required to arm") but that is perfectly fine and probably covers 99.9% of use cases. Thank you!

**Reply by edwardtfn** [reply]
> Indeed you implemented it only for arm and not for disarm - I can see why (HA only exposes "code required to arm") but that is perfectly fine and probably covers 99.9% of use cases.

That's right, I only have from Home Assistant if a code is required and if a code is required to arm, so I would expect if a code is required, but not for arming, that should be required for disarming, isn't that a right conclusion?

> You are faster than our combined ability to come up with suggestions! Amazing

😂

**Reply by Doughboy68** [reply]
The way it works for me in home assistant is if the code is entered in the alarm config, then HA never displays the keypad panel and I can arm/disarm just by pressing the button in Home Assistant. It will therefore use that code from config automatically.

I prefer to leave the code out of the config so the alarm panel appears, having to enter a code each time. When I arm the panel from HA, I don't have to enter a code and can just press ARM HOME or ARM AWAY. In  NSPanel, the keypad pops up when I press an arm mode, and I can just press the Check mark in lower right to arm without a code, probably because the keypad is not displayed all the time and the way it is coded... not a big deal, just comparing to how home assistant works. All alarms I have used do not require a code to arm, and will arm just pressing a button eg. home, away, etc.  Does anyone else have any input on this... is there an alarm out there which requires a code to be entered to arm it?

**Reply by edwardtfn** [reply]
Could you please let me know the values of attributes in your alarm control panel entity?

**Reply by Doughboy68** [reply]
code_format: number
changed_by: null
code_arm_required: true
friendly_name: Home Alarm
supported_features: 15

Even though is says code_arm required, I do not need a code to arm. Maybe this is just my envisalink since I can arm the alarm also by pressing/holding button for arm on the panel mounted on wall.

What I was meaning is that the alarm panel in HA looks like this and I can just press the ARM HOME or ARM AWAY. But with the NSPanel, the button to actually arm is in the keypad that pops up after selecting an arming mode, so I just press lower right button ignoring entering code. It works fine this way. It does require a code to disarm.

![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/16811687/9fdd14c3-5952-4b1c-a823-332562518625)

**Comment by doron1** [comment]
Here is another, somewhat far-fetched but very useful IMO, idea for improving the alarm system code.
The use case is for when the NSPanel is installed at the main entrance door.

When the door is opened (or a similar entry sensor is triggered), the alarm typically moves into a "pending" state, which is in fact the grace time you are given to disarm, before it moves to "triggered". This is usually a 20-45 sec delay during which you need to find the physical alarm panel, remember the disarm code and key it in. Typically, during this time there'd also be a repeating beep, reminding you to disarm.

So: How about an _optional_ setting ("primary alarm panel"?). If this option is set, then, when the alarm system moves to the pending state, the code will
(a) "take over" the panel by moving to the alarm panel and pop up the disarm keyboard
(b) beep every 2 seconds
When the alarm moves to disarmed or any armed state, the "hijack" is released. (If it moves to "triggered", I guess it would stay hijacked? TBD)

I did say this is a bit far-fetched :-)  But when the alarm is an important part of a particular NSPanel use, this will be quite useful.

**Comment by Doughboy68** [comment]
I made some modifications to TFT and "borrowed" an entity page and a button page to show what I mean for displaying my alarm zone entities, and bypassing them. I just made it swipe between them.

With this test, I am able to swipe from alarm page left/right to show the status of my alarm keypad and any sensors - the one I test is my garage door, and button page to bypass zones. I bypass the garage door open zone to show that the physical keypad shows bypass. If the door was opened, it would still show green checkmark LED and be in ready state.


Sorry for bad video, was a test panel I plugged in and trying to do with one hand, plus I had just woke up so I am pressing some wrong buttons and having difficulty swiping LOL - wanted to do this before heading out to work! 

https://1drv.ms/v/s!Apx8MNJLNAiagvU3ZKpYSGNky1GT4w?e=HUOCsB

**Reply by edwardtfn** [reply]
It looks cool.

Is there a way to get from the system the list of custom zones in a programmatically way? I mean, is that info stored somewhere in your alarm integration in Home Assistant?
Can you share your alarm attributes when custom bypass is active? Please remove any sensitive info before sharing.

**Reply by Doughboy68** [reply]
The alarm integration I am using is envisalink_new which is still in HACS but is working towards refactoring and going back in the core (the core version does not have bypass ability).
https://github.com/ufodone/envisalink_new

I checked the alarm attributes and there is no changes when I do bypass. I believe that this integration handles bypassing manually based on the way the panel works and makes entities for them, whereas alarms designed for home assistant may have a custom bypass feature built-in but I am not sure how that works and wasn't able to find much info about its usage.

Here are entities created...
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/16811687/9c7cf148-55ab-4230-ac31-074b110f225c)

These bypass entities are created optionally in integration and I renamed all of the sensors and bypass switch suffix to be more meaningful.
You could populate them based on the "Home Alarm" name in my case... binary_sensor.{alarmname}_alarm_zone_....  and switch.{alarmname}.alarm_zone_... and the keypad is just sensor.{alarmname}_keypad

I think that this would require specific handling of this envisalink integration, considering other alarm panels which exist may work different then more information is needed about bypass for different alarms to see how it is done. Maybe after this release is made into the main version, more people will provide feedback how their zones and bypass entities work.

Also, off topic here, but I did notice that while sitting on the home page, the alarm icon does not update on its own if alarm happens to change elsewhere... have to go into alarm page and exit for the home page icon to update.

**Reply by edwardtfn** [reply]
> Also, off topic here, but I did notice that while sitting on the home page, the alarm icon does not update on its own if alarm happens to change elsewhere... have to go into alarm page and exit for the home page icon to update.

I will take a look at that.

**Reply by edwardtfn** [reply]
I believe I found a fix for that. Already applied to `dev`. You will have to update ESPHome and Blueprint. No need to update TFT this time.

**Reply by Doughboy68** [reply]
Tried it and it is fixed somewhat if I go into any other page and back, but it never refreshes when the display is not touched.

**Reply by edwardtfn** [reply]
Ok, I will have to work more on this.

**Reply by edwardtfn** [reply]
Now I could test and found the bug. Already fixed on `dev`.
This time you have to update ESPHome only.

**Reply by Doughboy68** [reply]
Tested and the icon now updates correctly.

**Comment by edwardtfn** [comment]
My plan right now is to finish the last bugs we can find with the current implementation and then release, so we take new functionalities in the following release.
Please let me know your thoughts.

**Reply by Doughboy68** [reply]
I have been thinking about this, and what about the following changes to alarm page?

The layout does not seem to fit in the style as other pages. The buttons are larger etc. Therefore would have to be redesigned a bit for the following:

- add sensors similar to climate screen
- add at least two custom buttons similar to climate screen

This would allow, for me anyways, flexibility to customize a bit. It may be beneficial to others for other reasons too, depending on their alarm system. They are added to climate screen, so would see no harm to add these to alarm screen also.

In my case I would use my alarm keypad sensor to show if my alarm is in a ready to arm state or not (not the same as alarm status eg. arm, disarm etc.). The two custom buttons I would use to call scripts to bypass or turn off bypass for all open zones (rather than worry about individual zone buttons on nspanel). Or these could be used for other things, or not used at all... up to the user. In my test I am not just putting these two buttons on a regular page, I use the confirm execution feature, and my script also sends a message to the panel confirming to user that the action was done and works quite well - seems can put a lot of text in a notification so could use a template that shows zones that were bypassed etc.

**Reply by edwardtfn** [reply]
Thanks again for your feedback.

> The layout does not seem to fit in the style as other pages. The buttons are larger etc.

I've copied the buttons from buttons pages, so the design and size are the same as there, but I agree that might not be the best. and we probably can reduce those a bit in order to fit a couple of other buttons there.

I'm thinking also about removing the text from the buttons (just like in many other places) so I don't have to handle translations, although they probably will have to be bigger icons, as this page is all about those.

As it is implemented now, there's no room for anything else on that page. Adding a couple of buttons is quite easy from coding point of view, but find a place to display those in the screen is quite challenging.

I will take a look.

**Reply by Doughboy68** [reply]
The buttons on alarm page are spaced out further apart. There are only six and the buttons page there are eight. If you rearranged there would be room for two more and squeeze in a couple sensors somewhere maybe at top or bottom.

**Comment by edwardtfn** [comment]
It's out for `beta` now: #1046

**Comment by edwardtfn** [comment]
I'm closing this discussion as we have released v4.0 (#1057).
Please add a feature request for the missing things.

And thanks a lot for your support.

**Comment by plasterer41** [comment]
Love it. Great work Sent from my iPhoneOn 18 Sep 2023, at 18:39, Edward Firmo ***@***.***> wrote:﻿
It's out for beta now: #1046
